read_test.go

```diff
@@ -595,6 +595,78 @@ func TestReadPixelData_SkipPixelData(t *testing.T) {
+		0x02, 0x00, 0x10, 0x00, 0x55, 0x49, 0x14, 0x00, 0x31, 0x2e, 0x32, 0x2e, 0x38, 0x34, 0x30, 0x2e,
+		0x31, 0x30, 0x30, 0x30, 0x38, 0x2e, 0x31, 0x2e, 0x32, 0x2e, 0x35, 0x00, 0x02, 0x00, 0x12, 0x00,
+		0x55, 0x49, 0x20, 0x00, 0x31, 0x2e, 0x36, 0x2e, 0x36, 0x2e, 0x31, 0x2e, 0x34, 0x2e, 0x31, 0x2e,
+		0x39, 0x35, 0x39, 0x30, 0x2e, 0x31, 0x30, 0x30, 0x2e, 0x31, 0x2e, 0x30, 0x2e, 0x31, 0x30, 0x30,
+		0x2e, 0x34, 0x2e, 0x30, 0x08, 0x00, 0x18, 0x00, 0x55, 0x49, 0x2c, 0x00, 0x31, 0x2e, 0x33, 0x2e,
+		0x36, 0x2e, 0x31, 0x2e, 0x34, 0x2e, 0x31, 0x2e, 0x33, 0x35, 0x31, 0x39, 0x30, 0x2e, 0x34, 0x2e,
+	}
+	data = append(preamble, data...)
+	dcm_header := bytes.NewBuffer(data)
+	return dcm_header
+}
+
+// Returns a fake DICOM group 2 header with a FileMetaInformationGroupLength tag (0x0002,0x0000)
+func fakeHeader_WithFileMetaInformationGroupLength() *bytes.Buffer {
+	preamble := make([]byte, 128)
+	// First element after DICM is (0x0002,0x0000)
+	data := []byte{
+		0x44, 0x49, 0x43, 0x4d, 0x02, 0x00, 0x00, 0x00, 0x55, 0x4c, 0x04, 0x00, 0xbc, 0x00, 0x00, 0x00,
+		0x02, 0x00, 0x01, 0x00, 0x4f, 0x42, 0x00, 0x00, 0x02, 0x00, 0x00, 0x00, 0x00, 0x01, 0x02, 0x00,
+		0x02, 0x00, 0x55, 0x49, 0x1a, 0x00, 0x31, 0x2e, 0x32, 0x2e, 0x32, 0x37, 0x36, 0x2e, 0x30, 0x2e,
+		0x37, 0x32, 0x33, 0x30, 0x30, 0x31, 0x30, 0x2e, 0x33, 0x2e, 0x31, 0x2e, 0x30, 0x2e, 0x31, 0x00,
+		0x02, 0x00, 0x03, 0x00, 0x55, 0x49, 0x2c, 0x00, 0x31, 0x2e, 0x33, 0x2e, 0x36, 0x2e, 0x31, 0x2e,
+		0x34, 0x2e, 0x31, 0x2e, 0x33, 0x35, 0x31, 0x39, 0x30, 0x2e, 0x34, 0x2e, 0x31, 0x2e, 0x32, 0x30,
+		0x32, 0x31, 0x30, 0x36, 0x30, 0x38, 0x2e, 0x36, 0x30, 0x37, 0x37, 0x33, 0x33, 0x35, 0x34, 0x39,
+		0x35, 0x39, 0x33, 0x00, 0x02, 0x00, 0x10, 0x00, 0x55, 0x49, 0x14, 0x00, 0x31, 0x2e, 0x32, 0x2e,
+		0x38, 0x34, 0x30, 0x2e, 0x31, 0x30, 0x30, 0x30, 0x38, 0x2e, 0x31, 0x2e, 0x32, 0x2e, 0x35, 0x00,
+		0x02, 0x00, 0x12, 0x00, 0x55, 0x49, 0x1c, 0x00, 0x31, 0x2e, 0x32, 0x2e, 0x32, 0x37, 0x36, 0x2e,
+		0x30, 0x2e, 0x37, 0x32, 0x33, 0x30, 0x30, 0x31, 0x30, 0x2e, 0x33, 0x2e, 0x30, 0x2e, 0x33, 0x2e,
+		0x36, 0x2e, 0x37, 0x00, 0x02, 0x00, 0x13, 0x00, 0x53, 0x48, 0x10, 0x00, 0x4f, 0x46, 0x46, 0x49,
+		0x53, 0x5f, 0x44, 0x43, 0x4d, 0x54, 0x4b, 0x5f, 0x33, 0x36, 0x37, 0x20, 0x08, 0x00, 0x18, 0x00,
+	}
+	data = append(preamble, data...)
+	dcm_header := bytes.NewBuffer(data)
+	return dcm_header
+}
+
+func TestReadHeader_TryAllowErrorMetaElementGroupLength(t *testing.T) {
+	opts := parseOptSet{allowErrorMetaElementGroupLength: true}
+	dcmheader_noinfogrplen := fakeHeader_NoFileMetaInformationGroupLength()
+
+	r := &reader{
+		rawReader: dicomio.NewReader(bufio.NewReader(dcmheader_noinfogrplen), binary.LittleEndian, int64(dcmheader_noinfogrplen.Len())),
+		opts:      opts,
+	}
+	_, err := r.readHeader()
+	if err != nil {
+		t.Errorf("unsuccesful readHeader when parse option %v is turned on and header has no MetaElementGroupLength tag", opts.allowErrorMetaElementGroupLength)
+	}
+
+	// -------------
```

Comment:
no need for this comment, though you may consider seperating the two cases into two seperate test functions or can use a [subtest](https://pkg.go.dev/testing#hdr-Subtests_and_Sub_benchmarks)

Reply:
Sounds good...